src/main/java/parking/services/PricingService.java

```diff
@@ -0,0 +1,42 @@
+package parking.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import parking.entities.DateContainer;
+import parking.entities.Plate;
+
+import java.rmi.activation.UnknownObjectException;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Collection;
+import java.util.Optional;
+
+@Service
+public class PricingService {
+
+    private final Collection<AbstractRegionPricing> abstractRegionPricing;
+    private final PlateManagerService plateManagerService;
+
+    @Autowired
+    public PricingService(Collection<AbstractRegionPricing> abstractRegionPricing, PlateManagerService plateManagerService) {
+        this.abstractRegionPricing = abstractRegionPricing;
+        this.plateManagerService = plateManagerService;
+    }
+
+    public void updatePrice(Plate plate) throws UnknownObjectException {
+
+        Optional<AbstractRegionPricing> searchResult = abstractRegionPricing.parallelStream()
```

Comment:
Czemu używasz parallelStream?

Reply:
Jedno źródło mówi, że powinno się stosować parallel stream kiedy się da, no bo w końcu wykorzystajmy te x rdzeni, ale jak teraz przeglądam internet to radzi się, żeby pozostawać po prostu przy stream kiedy nie mamy za wiele danych do przejrzenia. Tak jak i w tym przypadku. No ale hej - wiecie, że ja wiem o parallel stream.